### PR TITLE
Add test for `internal/db/storage/blob/s3/reader`.

### PR DESCRIPTION
**Pull request by hlts2**
Signed-off-by: hlts2 <hiroto.funakoshi.hiroto@gmail.com>

### Description:

I refactored and added test for`s3/reader.go`

#### Refactor

- WHY
I am writing test about s3/reader now. But There were some places where I couldn’t write a test.
The reason are as follows.
1. The current implementation calls New method  inside test target function. https://github.com/vdaas/vald/blob/92f7fff4acdef195c0a8e62ac981fc762dbc78e0/internal/db/storage/blob/s3/reader/reader.go#L97

2. The current implementation use s3 struct object, not interface https://github.com/vdaas/vald/blob/92f7fff4acdef195c0a8e62ac981fc762dbc78e0/internal/db/storage/blob/s3/reader/reader.go#L40

- WHAT

1. I want to create io interface to create object via interafce like comress. https://github.com/vdaas/vald/blob/master/internal/compress/lz4/lz4.go#L49
2. I want to use s3iface.S3API interface instead of s3 struct(*s3.S3) 
https://github.com/aws/aws-sdk-go/blob/v1.34.28/service/s3/s3iface/interface.go#L62-L441

```go
type reader struct {
	service *s3.S3
}
↓
type reader struct {
	service s3iface.S3API
}
```

### Environment:

- Go Version: 1.15.2
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.1

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [x] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: internal/db/storage/blob/s3/reader/reader.go [hunk 1]
```diff
@@ -26,34 +26,39 @@ import (
...
 	"github.com/vdaas/vald/internal/errors"
-	ctxio "github.com/vdaas/vald/internal/io"
 	"github.com/vdaas/vald/internal/log"
 	"github.com/vdaas/vald/internal/safety"
 )
 
 type reader struct {
 	eg      errgroup.Group
-	service *s3.S3
+	service s3iface.S3API
```
Comment (kevindiu) [inline internal/db/storage/blob/s3/reader/reader.go:41]: why the package name changed?

Comment (hlts2) [reply internal/db/storage/blob/s3/reader/reader.go:41]: I wrote the this reason.
could you please read pr description?

File: internal/db/storage/blob/s3/reader/reader_mock.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
+package reader
```
[the remark this answers is not in the document]
Comment (hlts2) [reply internal/db/storage/blob/s3/reader/reader_mock.go:1]: I think so too.
but we should move the mock of writer pacakge too.
so after this PR is merged, I will move it.

File: internal/db/storage/blob/s3/reader/reader.go (hunk 1 above)
Comment (kevindiu) [reply internal/db/storage/blob/s3/reader/reader.go:41]: sorry I am not sure should we use this ?https://github.com/vdaas/vald/blob/d887e713544099356e59942429f3dc8d1378916f/internal/db/storage/blob/s3/sdk/s3/s3.go#L24

File: internal/db/storage/blob/s3/reader/reader_mock.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (kevindiu) [reply internal/db/storage/blob/s3/reader/reader_mock.go:1]: ok

**Review by kevindiu — APPROVED** [review]
LGTM

**Review by kevindiu — COMMENTED** [review]
please fix CI

**Review by kevindiu — APPROVED** [review]
LGTM

**Review by kevindiu — DISMISSED** [review]
LGTM

**Review by vankichi — DISMISSED** [review]

**Pull request merged**